Correct values in configuration on banning and retry

linked0 opened the pull request:
We should wait some time to ban any unanswering node after many retries and failed requests. We now set the waiting time to about 1 hour. Actually, the total time of multiplication of `retry_delay`, `max_retries`, `timeout`, and `max_failed_requests` in the commits could be one and a half hours.

We mostly refer to the values of `infrastructure` because it is used for the live test system. Anyway, one hour could be the best time to wait before banning a bad node.

Fixes #1657

linked0 committed:
Set default values for retry and banning

The configuration of retry and banning is more related than expected.
We should wait some time to ban any unanswering node after many retries
and failed requests. We set the waiting time to about 3 hours. Actually,
the total time of multiplication of `retry_delay`, `max_retries`,
`timeout`, and `max_failed_requests` could be one and a half hours. We
mostly refered to the values of `infrastructure` because it is used
for the live test system. Anyway, one hour could be the best time to
wait before banning a bad node.

linked0 committed:
Apply the default value to `TestConf`

There is no special reason to maintain other values for `retry_delay`,
`max_retries`, and `timeout` than the ones of `config.yaml`. But, Some network
tests could have other values like in `agora.test.NetworkDiscovery` module.

Geod24 commented on doc/config.example.yaml:
```diff
@@ -26,11 +26,11 @@ node:
   port:    2826    # 0xB0A, default value
   # Number of milliseconds to wait between retrying requests
-  retry_delay: 5000
+  retry_delay: 500
```
I think 5 seconds was actually better.

linked0 replied:
Will update.

linked0 replied:
@Geod24 I'd like to confirm that the value(`5000` msecs) you suggested is right before applying it to the code. I think it's kind of long. Do you mean we should have 5 seconds for `retry_delay` and 5 seconds for `timeout`?

linked0 replied:
@Geod24 I'd like to confirm that the value(5000 msecs) you suggested is right before applying it to the code. I think it's kind of long. Do you mean we should have 5 seconds for `retry_delay` and 5 seconds for `timeout`?

Geod24 replied:
Yes

Geod24 commented on source/agora/common/Config.d:
```diff
@@ -157,13 +157,13 @@ public struct NodeConfig
-    public Duration retry_delay = 3.seconds;
+    public Duration retry_delay = 500.msecs;
 
     /// Maximum number of retries to issue before a request is considered failed
-    public size_t max_retries = 5;
+    public size_t max_retries = 10;
 
     /// Timeout for each request
-    public Duration timeout = 500.msecs;
+    public Duration timeout = 5000.msecs;
```
Those should indeed be consistent with the example doc, see my comments there

linked0 replied:
I changed the value to `5000.msecs` in all the related files. Am I missing something?

kchulj replied:
So I assume we want 5 seconds for both `retry_delay` and `timeout`.

linked0 replied:
OK, I got it. I'm applying it now. Ty.

Geod24 commented on source/agora/test/Base.d:
```diff
@@ -1752,13 +1752,13 @@ public struct TestConf
 
     /// the delay between request retries
-    Duration retry_delay = 100.msecs;
+    Duration retry_delay = 500.msecs;
```
Those are slightly different though. We know that the nodes are in memory so it's going to be insanely faster.
However, increasing to 500 msecs (and reducing retries) is actually a good idea, so 👍

Geod24 commented on source/agora/test/NetworkDiscovery.d:
```diff
@@ -20,7 +20,10 @@ import agora.test.Base;
 unittest
 {
-    TestConf conf = TestConf.init;
+    TestConf conf =
+    {
+        retry_delay : 100.msecs,
+    };
```
Why ?

linked0 replied:
The first network test in this module would fail after the default `retry_delay` is set to `500.msecs` because the test checks the `NodeInfo` of all the nodes immediately.

Geod24 commented on doc/config.example.yaml:
```diff
@@ -117,7 +117,7 @@ flash:
 banman:
   # max failed requests until an address is banned
-  max_failed_requests: 1000
+  max_failed_requests: 100
```
On the fence on this, I think both are valid, since it's client requests.

Geod24 reviewed (approved):
Except for the default `retry_delay`, all LGTM

linked0 committed:
Make the `data_dir` consistent with config example

The original `data_dir` is good default for Linux but not for
Windows and OSX. So we changed this value to good default for
all the OSs and make the value consistent with the config example.

kchulj reviewed (approved):
✅

kchulj commented:
`Some index files failed to download. They have been ignored, or old ones used instead.` ?
There has been various `Integration` failures recently..

Geod24 commented:
What's up with the changes to flash in the 3rd commit ?

linked0 commented:
I will check the last commit. There might have been some problems in committing.

linked0 commented:
The 5 seconds for the `retry_delay` makes the integration test fail constantly. So I set the value to 3 seconds, which makes the test succeed. I will check the reason why it affects the test.
